### ISSUE TITLE
One hour limit before submitting Steam Controller configurations

### ISSUE THREAD

**Issue by major-gnuisance** [ValveSoftware/steam-for-linux#4268]
Currently, to submit a Steam Controller community configuration, it's required that the user plays one hour with it, before it can be published.

Good idea, terrible implementation.

Here's why it's terrible: *it severely punishes fine-tuning the configuration.*
And one hour is a *very long time.*
Any change to a configuration will reset the waiting period before the configuration can be submitted.

I just spent hours playing a game that's supposed to be played in small sessions (Race the Sun) so I could submit my configuration.

First, I had to nail the configuration.
I spent over an hour on this, but since I have to play with it for an hour *after the last modification,* I couldn't submit it.

So I begrudgingly began my one-hour-long wait, playing a game I was already tired of by this time.  
Then, about 40 minutes in, I noticed a problem with the configuration.

What could I do at that point?
A. Tweak the configuration and start over the waiting period.
B. Spend 20 more minutes and submit a defective configuration.
C. Give up.

I chose B, hoping I would be able to mend it after the submission.
So I wasted 20 more minutes, and uploaded the configuration.
And there it was, with the defect I had found, but couldn't fix without wasting 40 more minutes.

So I fixed the problem, and saw that I could indeed save over the one I had just uploaded without wasting another hour.
"Great," I though. So I did.

And now it's gone from the community configuration list.

Let's just say that I'm not very happy right now.

**Comment by not83** [comment]
Is this a bug specific to the steam for Linux client?

**Comment by major-gnuisance** [comment]
How should I know?
It's the only one I use.

**Comment by Tele42** [comment]
@not83 yes, this is part of the steam client, but there is a good chance it affects all platforms.

**Comment by Plagman** [comment]
Thanks for your feedback; this is by design, but we might tweak this in the future.

**Comment by major-gnuisance** [comment]
@Plagman was it supposed to remove the configuration the community list when I tried to update it?

**Comment by Plagman** [comment]
Saving over a configuration just deletes the previous one and creates a new one. It might take a little bit for the new one to show up due to caching issues, has it still not re-surfaced?

**Comment by major-gnuisance** [comment]
No, it never came back.
Meanwhile I uploaded two configurations for that game, which I can see on the list.

I would've kept it a single configuration, but I never tried updating again, out of fear it might vanish from the list like the first one.

By the way: I might have written over the first configuration twice in quick succession before it vanished.
I can't quite remember.

I'd try to reproduce it, but I don't want to waste my time waiting for an arbitrary counter to tick down.

**Comment by major-gnuisance** [comment]
In any case, the whole concept of the time limit needs to be revised, if it's to be kept.
FWIW, I'd suggest that it were based on two separate variables:
- Total time spent playing the game with a Steam Controller (long-ish, ~1 hour)
- Time spent playing with a particular configuration (short, ≤10 minutes)
Minor changes such as labeling buttons shouldn't reset the time limit.

Tough these limits will still be a grievance for someone going through a series of games with identical controls, such as visual novels or point-and-click adventures.

Another thing: users should get at least coarse feedback on how much more time they need to play before submitting.
"Please play %d more minutes with your configuration before publishing it."
Not "Please play with your configuration for at least an hour before publishing it." as it is right now.

**Comment by goatofdeath** [comment]
I'm also running into time limit issues with submitting a config. One thing that really sucks is I can't upload an older configuration I made well before the time limit was introduced without playing an hour. But even worse is the time tracking just plain doesn't seem to work for me. At least not with the Steam client on Windows. I played about two hours tonight to be able to submit my Borderlands PreSequel config and I still can't submit it. Sucks to say the least. I want to give to the community. However, I have better things to do with my time than work around badly implemented limits set by Steam that reduce my ability to share with others.
